include/ensmallen_bits/cmaes/cmaes.hpp

```diff
@@ -70,8 +75,8 @@ class CMAES
    *     objective.
    */
   CMAES(const size_t lambda = 0,
-        const double lowerBound = -10,
-        const double upperBound = 10,
+        const TransformationPolicyType& 
+              transformationPolicy = TransformationPolicyType(),
```

Comment:
This breaks reverse compatibility---if the default transformation policy type is the bounding box (which takes parameters `lowerBound` and `upperBound` anyway), one option could be to preserve a deprecated constructor that takes `lowerBound` and `upperBound` and throws an exception if `TransformationPolicyType` is not `BoundaryBoxConstraint`.  We could remove that deprecated constructor for ensmallen 3.x, but for now we would need to preserve it so we satisfy the guarantees in `UPDATING.txt`.

We would also need to update the documentation to reflect the new API.

Reply:
The default transformation policy is `EmptyTransformation`. However, I could preserve a constructor that takes `lowerBound` and `upperBound` as parameters of type `double`. This would then throw an exception if `TransformationPolicyType` is not `BoundaryBoxConstraint` like you said.

Reply:
Yeah, that would be nice. Ideally this should preserve reverse compatibility completely, so that if a user still used the old constructor, the behavior would be identical.  That might result in some amount of slightly ugly code, but you can mark the relevant functions as deprecated and leave a comment that they should be removed for the next major version of ensmallen. :+1: